src/scanpy/_compat.py

```diff
@@ -90,3 +96,93 @@ def pkg_version(package: str) -> Version:
     # but this code makes it possible to run scanpy without it.
     def old_positionals(*old_positionals: str):
         return lambda func: func
+
+
+@overload
+def njit(fn: Callable[P, R], /) -> Callable[P, R]: ...
+@overload
+def njit() -> Callable[[Callable[P, R]], Callable[P, R]]: ...
+def njit(
+    fn: Callable[P, R] | None = None, /
+) -> Callable[P, R] | Callable[[Callable[P, R]], Callable[P, R]]:
+    """Jit-compile a function using numba.
+
+    On call, this function dispatches to a parallel or sequential numba function,
+    depending on if it has been called from a thread pool.
+
+    See <https://github.com/numbagg/numbagg/pull/201/files#r1409374809>
+    """
+
+    def decorator(f: Callable[P, R], /) -> Callable[P, R]:
+        import numba
+
+        fns: dict[bool, Callable[P, R]] = {
+            parallel: numba.njit(f, cache=True, parallel=parallel)  # noqa: TID251
+            for parallel in (True, False)
+        }
```

Comment:
Does this jit twice?

```python
def foo():
    print('bar')
    return 1 
d = { k: foo() for k in ['a', 'b'] }
#bar
#bar
```

Reply:
I don’t know what you mean. jit means “just in time” so like every regular numba-decorated function, it’ll eagerly create the wrapper (in this case two wrappers) and jit-compile once you call the wrapper the first time with a new set of input types.

Reply:
You're right, hadn't considered the "jit" aspect of things.

Reply:
I see! I was really confused haha!